experiments/ClimaEarth/setup_run.jl

```diff
@@ -231,6 +232,22 @@ function setup_and_run(config_dict::AbstractDict)
     Random.seed!(random_seed)
     @info "Random seed set to $(random_seed)"
 
+    isnothing(restart_t) && (restart_t = Checkpointer.t_start_from_checkpoint(dir_paths.checkpoints))
+    isnothing(restart_dir) && (restart_dir = dir_paths.checkpoints)
+    if !isnothing(restart_t)
```

Comment:
Should this be
```suggestion
    if should_restart
```
where `should_restart = !isnothing(restart_t) && !isnothing(restart_dir)` is defined lower